src/fakeauth.js

```diff
@@ -0,0 +1,54 @@
+var debug = require('debug')('taskcluster-lib-testing:FakeAuth');
+var nock = require('nock');
+var hawk = require('hawk');
+
+/**
+ * Intercept requests to the auth service's authenticateHawk method and
+ * return a response based on clients, instead.  This is useful when testing
+ * other API services.  Note that accessTokens are not checked -- the fake
+ * simply controls access based on clientId or the scopes in a temporary
+ * credential or supplied with authorizedScopes.
+ *
+ * Clients is on the form
+ * ```js
+ * {
+ *  "clientId1": ["scope1", "scope2"],
+ *  "clientId2": ["scope1", "scope3"],
+ * }
+ *
+ * Call `stop` in your test's `after` method to stop the HTTP interceptor.
+ */
+module.exports.start = function(clients) {
```

Comment:
Just `exports.start` is sufficient...

Just an idea.. but you can technically do: `require('mocha').afterEach(() => {exports.stop();});`
That should clear it after each test case...

See example:
https://github.com/taskcluster/taskcluster-lib-testing/blob/106ea245cfa18e477aeecf4d639261a7d2456cfe/src/testing.js#L86

We should probably make `mocha` a `peerDependency` of this and `taskcluster-base`, for this to always work... But it does generally work... An additional trick I use is the pass mocha to `start(clients, mocha)`. That way you have the mocha instance used in the tests.